### PR TITLE
lock cysync when desktop goes to sleep

### PR DESCRIPTION
**Pull request by HarishRocks**

### CODE REVIEW

**Comment by irshadCypherock** [comment]
Task linked: [CSI-1015 Lock CySync on sleep](https://app.clickup.com/t/37308523/CSI-1015)

**Comment by HarishRocks** [comment]
@irshadCypherock ready to merge

**Comment by HarishRocks** [comment]
@irshadCypherock added the addtional check like you suggested

File: cysync/src/store/provider/lockscreenProvider.tsx [hunk 1]
```diff
@@ -40,26 +41,53 @@ export const LockscreenProvider: React.FC = ({ children }) => {
...
-  }, [lockscreen, isPasswordSet, autolockTime]);
+  };
+
+  useEffect(() => {
+    ipcRenderer.on('lock-screen', onDesktopLock);
+
+    return () => {
+      ipcRenderer.removeListener('lock-screen', onDesktopLock);
+    };
+  }, [lockscreen, isPasswordSet, autoLock]);
```
Comment (irshadCypherock) [inline cysync/src/store/provider/lockscreenProvider.tsx:85]: The arguments are not needed if we add the listener every time.

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**